src/python/pants/engine/internals/scheduler.py

```diff
@@ -75,6 +75,13 @@
 Workunit = Dict[str, Any]
 
 
+@dataclass(frozen=True)
+class CurrentGoals:
+    """The currently running goals."""
+
+    rule_names: tuple[str, ...]
```

Comment:
Should this be rule names, or the `Goal.subsystem.name`?

Reply:
I'd prefer the `Goal.subsystem.name` but failed to figure out how to get that :P 
Then again, I realized this may be more robust, or too rigid I suppose, depending on the use case. Maybe it would be nice to capture both?